scripts/remoteperformance.ps1

```diff
@@ -0,0 +1,298 @@
+        } else {
+            $Session = New-PSSession -ComputerName $ComputerName -ConfigurationName PowerShell.7
+        }
+    } else {
+        $Session = New-PSSession -HostName "$Remote"
+    }
+
+    $RemoteAddress = $Session.ComputerName
+
+    if ($null -eq $Session) {
+        Write-Error "Failed to create remote session"
+        exit
+    }
+
+    $LocalAddress = Get-LocalAddress -RemoteAddress $RemoteAddress
+
+    Write-Output "Connected to: $RemoteAddress"
+    Write-Output "Local IP Connection $LocalAddress"
+}
+
+Set-Session -Session $Session
+
+$OutputDir = Join-Path $RootDir "artifacts/PerfDataResults"
+New-Item -Path $OutputDir -ItemType Directory -Force | Out-Null
+
+$RemotePlatform = Invoke-TestCommand -Session $Session -ScriptBlock {
+    if ($IsWindows) {
+        return "windows"
+    } else {
+        return "linux"
+    }
+}
+
+# Join path in script to ensure right platform separator
+$RemoteDirectory = Invoke-TestCommand -Session $Session -ScriptBlock {
+    Join-Path (Get-Location) "Tests"
+}
+
+$LocalDirectory = Join-Path $RootDir "artifacts"
+
+if ($Local) {
+    $RemoteDirectory = $LocalDirectory
+}
+
+$CurrentCommitHash = Get-GitHash -RepoDir $RootDir
+
+function LocalSetup {
+    $RetObj = New-Object -TypeName psobject
+    $RetObj | Add-Member -MemberType NoteProperty -Name apipaInterfaces -Value $null
+    try {
+        if ($IsWindows) {
+            $apipaAddr = Get-NetIPAddress 169.254.*
+            if ($null -ne $apipaAddr) {
+                # Disable all the APIPA interfaces for URO perf.
+                Write-Debug "Temporarily disabling APIPA interfaces"
+                $RetObj.apipaInterfaces = (Get-NetAdapter -InterfaceIndex $apipaAddr.InterfaceIndex) | Where-Object {$_.AdminStatus -eq "Up"}
+                $RetObj.apipaInterfaces | Disable-NetAdapter -Confirm:$false
+            }
+        }
+    } catch {
+        $RetObj.apipaInterfaces = $null
+    }
+
+    return $RetObj
+}
+
+function LocalTeardown {
+    param ($LocalCache)
+    if ($null -ne $LocalCache.apipaInterfaces) {
+        # Re-enable the interfaces we disabled earlier.
+        Write-Debug "Re-enabling APIPA interfaces"
+        $LocalCache.apipaInterfaces | Enable-NetAdapter
+    }
+}
+
+function Invoke-Test {
+    param ($Test)
+
+    Write-Output "Running Test $Test"
+
+    $RemoteExe = Get-ExeName -PathRoot $RemoteDirectory -Platform $RemotePlatform -IsRemote $true -TestPlat $Test.Remote
+    $LocalExe = Get-ExeName -PathRoot $LocalDirectory -Platform $LocalPlatform -IsRemote $false -TestPlat $Test.Local
+
+    # Check both Exes
+    $RemoteExeExists = Invoke-TestCommand -Session $Session -ScriptBlock {
+        param ($RemoteExe)
+        Test-Path $RemoteExe
+    } -ArgumentList $RemoteExe
+
+    $LocalExeExists = Test-Path $LocalExe
+
+    if (!$RemoteExeExists -or !$LocalExeExists) {
+        Write-Output "Failed to Run $Test because of missing exe"
```

Comment:
Make this a `Write-Error` and move this down and replace the `return` with it. Generally, replace all the `Write-Output` for error messages with `Write-Error` so things immediately terminate.